zaberMotionSupport/Makefile

```diff
@@ -0,0 +1,17 @@
+TOP=..
+include $(TOP)/configure/CONFIG
+#----------------------------------------
+#  ADD MACRO DEFINITIONS AFTER THIS LINE
+#=============================
+
+ifndef ZML_HOST_ARCH
+$(error "motorZaberMotion: Unsupported architecture: $(EPICS_HOST_ARCH)")
+endif
+
+LIB_INSTALLS += $(wildcard ../ZaberMotionCppSupport/lib/$(ZML_HOST_ARCH)/*)
```

Comment:
﻿>I was able to further simplify the build configuration using a relative path to the include files. I pushed my changes to the review-branch-build-improvements branch:
https://github.com/colbysparks/motorZaberMotion/compare/review-branch...review-branch-build-improvements?expand=1

Apparently there were a lot of cascading effects from my not having specified the `LIB_INSTALLS` path correctly. Now that this has been updated, the support libs always get installed to the correct location, so there is no longer an need to specify `-rpath` nor add any extra `-L` paths.

Thanks for figuring this out! When I saw other repos using `../` to install support libs I thought they might have been, but it turns out that this path needs to be relative to the Makefile in `./O.linux-<arch>`, not relative to this file path..

I just tested your changes on `review-branch-build-improvements` on my linux machine and everything works, but I'm getting a runtime linking error running the example ioc on macOS. I'll commit these changes once I figure it out.

Reply:
I'm sorry I don't have an OS X machine I can use for testing.

Reply:
All good. I just discovered that the build system breaks symlinks when it copies the libs from the support package folder. I'll have to make some changes so that it doesn't just add three copies of the same library to the target lib install folder.

That said, even if `libzaber-motion.dylib` is a copy of the versioned lib it points to, runtime linking should still work on macOS so I doubt that's the issue I'm seeing.

<img width="1084" alt="image" src="https://github.com/user-attachments/assets/284dce81-9a62-4906-9a34-0153a5425e7d">

Reply:
I haven't managed to find a good solution to the symlink copying issue in the EPICS build system, nor on the gulpfile scripting side of things. I might just have to make changes to how the zml support package is built, but that's a conversation I need to have with my team. I am fine leaving this for now but will make a note to fix some time in the short term.

As for the macOS problem, it's very likely that I could have missed something, but it seems like there may be an oversight in the EPICS build system where rpaths are not included for shared libs which need them to be specified. When linking the example ioc on linux (and `LINKER_USE_RPATH=YES`), the build system system will add rpaths for all the lib folders, including the folder which contains the support libs:
```
-Wl,-rpath,/home/ubuntu/EPICS/support/motor/lib/linux-aarch64
```
But for darwin builds no rpaths are passed to the linker, even with `LINKER_USE_RPATH=YES`, so it looks like a macOS user may have to manually add the rpath or use a command line tool for specifying an rpath in an existing executable.

Reply:
The "-rpath" option for Linux comes from this config file in EPICS base:

https://github.com/epics-base/epics-base/blob/b7cc33c3c9c1a6cc14290b9a558e97bd89171e80/configure/os/CONFIG.Common.linuxCommon#L26-L27

There isn't any reference to "rpath" in any of the darwin config files:

```
❯ pwd
/home/username/epics/local/zaberReview/base-7.0.8.1/configure/os
❯ grep rpath * | grep -i darwin
❯ 
```

I don't know if it is an oversight or not.  The EPICS base developers have a mailing list where they answer questions like this:

https://epics.anl.gov/core-talk/

Reply:
>The "-rpath" option for Linux comes from this config file in EPICS base:
https://github.com/epics-base/epics-base/blob/b7cc33c3c9c1a6cc14290b9a558e97bd89171e80/configure/os/CONFIG.Common.linuxCommon#L26-L27

I ended up finding this yesterday shortly after making that comment. Adding the following to either `CONFIG` or `CONFIG_SITE` does fix the build issue with darwin:

```
PRODDIR_RPATH_LDFLAGS_YES += $(PROD_DEPLIB_DIRS:%=-Wl,-rpath,%)
PRODDIR_LDFLAGS += $(PRODDIR_RPATH_LDFLAGS_$(LINKER_USE_RPATH))
```

Reply:
I just emailed the core-talk mailing list, so I will wait to hear back from them on whether or not the above is an appropriate solution for darwin builds

Reply:
I heard back from the core talk forum and apparently my solution is fine, so I went and updated the build config and documentation for darwin users. I've also made a few other changes to improve error handling and reporting, and implemented `asynMotorAxis::setPosition`. 

I am starting to feel like this is ready for release. If you have any other comments or concerns, please let me know. Otherwise, thanks so much for taking the time to do this review. Your help has been invaluable.